### PR TITLE
Golang Core Dev Team - August 13, 2018

### PR DESCRIPTION
**Pull request by keks**
The link to the recording is not there yet.

### CODE REVIEW

**Comment by djdv** [comment]
I refined my notes a bit.

**Comment by djdv** [comment]
@schomatis 
Added.

**Comment by diasdavid** [comment]
@schomatis gave you access as well for future times!

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+   - none
+ - Next:
+   - Continue doing the Go Core Dev Team Interviews
+   - Encourage everyone to do some Waffle Grooming
+     - rough time scale: 30min/wk
+
+
+@marten-seemann (not attending)
+ - Done: 
+   - various QUIC improvements
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: @marten-seemann, mind enumerating here what improvements were made?

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+### Week Update
+
+@stebalien
+ - Done:
+   - Reviewed a bunch of PRs
+   - Responded to/triaged a bunch of issues
+   - Fixed a few bugs
+   - Notable: update gogo protobuf
+ - Blocked:
+   - https://github.com/multiformats/multiaddr/pull/68 <- @daviddias P3
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: TODO: @diasdavid (myself)

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+## Notes
+
+### Week Update
+
+@stebalien
+ - Done:
+   - Reviewed a bunch of PRs
+   - Responded to/triaged a bunch of issues
+   - Fixed a few bugs
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Any notable bug (that people were eagerly waiting)?

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+   - Notable: update gogo protobuf
+ - Blocked:
+   - https://github.com/multiformats/multiaddr/pull/68 <- @daviddias P3
+   - https://github.com/multiformats/multiaddr/pull/68#issuecomment-401971665
+ - Next:
+   - More of the same
+     - CIDv1
+   - Finish the protocol negotiation spec
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: @Stebalien remind me, is this related with Multistream?

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+- next
+  - a few code reviews where i was summoned
+  
+@schomatis (not attending)
+ - Done:
+   - Reviewed a few PRs
+   - Studied part of the `gx`/`gx-go` code and workflow, how can we
+     make the package manager more user friendly?
+ - Next:
+   - Start submitting some PRs in `gx`/`gx-go` with documentation
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Does this mean you are fully focused on gx, @schomatis ?

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+
+@bigs
+ - Done:
+   - Research into virtual network topologies w/ openvswitch/iptools
+     -> @daviddias introduce travis & jacobheun
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Let's do Github Style!

@travisperson @jacobheun, please meet @bigs. I'm sure you have crossed each other or even met at the IPFS Dev Meetings.

- @bigs is doing a lot of IPTB work for libp2p testing
- @travisperson is doing a lot of IPTB work to add support for js-ipfs daemons (Node.js) and js-ipfs browser nodes
- @jacobheun is looking into designing interop tests that can be used by go-libp2p, js-libp2p and rust-libp2p

It would be great to have you all sync soon to make sure that we leverage everyone's focus and don't get duplicated work done.

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+   - DHT research group
+ - Blocked:
+   - Nothing
+ - Next:
+   - Durable implementation of namespaced virtual networking in go-netdef
+   - Write rendered network config to a file for easy reversal
+   - Learn virtual NATing?
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: If you end up writing notes, post them on libp2p/notes perhaps?

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+@djdv
+ - Done:
+    - Windows `ipfs mount` foundations (WinFSP, cgo-fuse)
+    - Read only`mount` support for /ipfs, /ipns, and /local(MFS)
+ - Blocked:
+    - https://github.com/golang/go/issues/18296
+      - Prevents use from building on non-Windows without cgo
+      - Having mutliple implimentations for the same subcommand seems non-ideal
+ - Next:
+   - Schedule to speak with David :^)
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: yes 👍  :)

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+  - Next:
+     - Continue Base32 CidV1 work
+     - Maybe add support for inlinling via the id-hash (https://github.com/ipfs/go-ipfs/pull/5281).
+    
+@lgierth (not attending)
+- done
+  - infra stuff, nothing go related
+- blocked
+  - nothing go related :)
+- next
```
Comment (jacobheun) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: @lgierth will you have bandwidth soon to pick https://github.com/ipfs/go-ipfs/pull/4595 back up?

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md (hunk 4 above)
Comment (Stebalien) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Yes. It's the "next" version of protocol negotiation.

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md (hunk 9 above)
Comment (lgierth) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Yes definitely until end-of-August -- how badly are you blocked?

Comment (jacobheun) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: I'm getting ready to do end to end validation, which needs the branch. I started implementing the endpoints in js, but it doesn't currently have refs support. I can look at getting the existing go branch running locally to at least get e2e validation happening. If we can get it live by mid September I think we can still get js peer/content delegated routing out by end of the quarter.

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md (hunk 4 above)
Comment (diasdavid) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Interesting. I didn't know that was urgent/`P0`

Comment (Stebalien) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: The sooner we fix it, the better. Every other protocol we have relies on it and we currently (in go at least) try rather hard to reuse streams as creating new ones is expensive (in terms of bandwidth).

For context, inefficient stream negotiation means we need things like https://github.com/libp2p/go-libp2p-kad-dht/pull/92 instead of https://github.com/libp2p/go-libp2p-kad-dht/pull/167.

File: meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+@stebalien
+ - Done:
+   - Reviewed a bunch of PRs
+   - Responded to/triaged a bunch of issues
+   - Fixed a few bugs
+   - Notable: update gogo protobuf
+ - Blocked:
+   - https://github.com/multiformats/multiaddr/pull/68 <- @daviddias P3
+   - https://github.com/multiformats/multiaddr/pull/68#issuecomment-401971665
+ - Next:
```
Comment (diasdavid) [inline meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: @Stebalien also, as discussed during the call. Please do the drawing (even if on paper) of something like:
https://github.com/ipfs/js-ipfs#code-architecture-and-folder-structure
image

But for go. Contemplating what is legacy, what is next, including:

- go-ipfs-cli
- go-ipfs-cmds
- go-ipfs-api
- go-ipfs-core
- go-ipfs daemon
  - go-ipfs-http-api
  - go-ipfs-http-gateway

And anything else that it is there.

Comment (diasdavid) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Bonus points if you do another for how things are pieced together. Like this one https://github.com/ipfs/js-ipfs#ipfs-core-architecture

Comment (diasdavid) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: **You can do it on paper**, I can create the ascii version

Comment (Stebalien) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Diagram:

commands

Source: https://gist.github.com/Stebalien/798f1d8d672f365ef1030fc810cea3ea

Comment (Stebalien) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Dotted means "likely going away". The "Legacy" parts are thin wrappers around some commands to translate between the new system and the old system. The grayed-out parts on the "daemon" diagram are there to show that the code is all the same, it's just that we turn some pieces on and some pieces off depending on whether we're running on the client or the server.

Comment (diasdavid) [reply meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md]: Added to the README with https://github.com/ipfs/go-ipfs/pull/5396
